emission/tests/modellingTests/TestGreedySimilarityBinning.py

```diff
@@ -24,31 +30,414 @@ def testBinning(self):
+
+        trips = etmm.generate_mock_trips(
+            user_id="joe", 
+            trips=n, 
+            origin=(0, 0), 
+            destination=(1, 1),
+            trip_part='od',
+            label_data=label_data, 
+            within_threshold=m, 
+            threshold=0.001,  # ~ 111 meters in degrees WGS84
+        )
+
+        # pass in a test configuration to the binning algorithm.
+        #
+        # clustering_way : Part of the trip used for checking pairwise proximity.
+        #                  Can take one of the three values:
+        #                  
+        #                   1. 'origin' -> using origin of the trip to check if 2 points
+        #                                   lie within the mentioned similarity_threshold_meters
+        #                   2. 'destination' -> using destination of the trip to check if 2 points
+        #                                       lie within the mentioned similarity_threshold_meters
+        #                   3. 'origin-destination' -> both origin and destination of the trip to check 
+        #                                             if 2 points lie within the mentioned 
+        #                                              similarity_threshold_meters
+        
+        model1_config = {
+            "metric": "od_similarity",
+            "similarity_threshold_meters": 111,  # meters,
+            "apply_cutoff": False,
+            "clustering_way": "origin",
+            "incremental_evaluation": False
+        }
+        model1 = eamtg.GreedySimilarityBinning(model1_config)
+        model1.fit(trips)
+
+
+        model2_config = {
+            "metric": "od_similarity",
+            "similarity_threshold_meters":111,  # meters,
+            "apply_cutoff": False,
+            "clustering_way": "destination",
+            "incremental_evaluation": False
+        }
+        model2 = eamtg.GreedySimilarityBinning(model2_config)
+        model2.fit(trips)
+
+
+        model3_config = {
+            "metric": "od_similarity",
+            "similarity_threshold_meters": 111,  # meters,
+            "apply_cutoff": False,
+            "clustering_way": "origin-destination",
+            "incremental_evaluation": False
+        }
+        model3 = eamtg.GreedySimilarityBinning(model3_config)
+        model3.fit(trips)
+
+        # Since both the origin and the destination points of the generated trips lie 
+        # within the threshold, there should be binning in all three configs.
+
+        at_least_one_large_bin = any(map(lambda b: len(b['feature_rows']) == m, model1.bins.values()))
+        self.assertTrue(at_least_one_large_bin, "at least one bin should have at least 5 features in it")
+
+        at_least_one_large_bin = any(map(lambda b: len(b['feature_rows']) ==m, model2.bins.values()))
+        self.assertTrue(at_least_one_large_bin, "no bin should have more than 1 features in it")
```

Comment:
the messages for the assert seem to be wrong given `len(b['feature_rows']) == m`

Reply:
True. Will fix.

Reply:
This was fixed as well.